packages/roosterjs-content-model-api/lib/modelApi/common/queryContentModelBlocks.ts

```diff
@@ -0,0 +1,91 @@
+import type {
+    ContentModelBlockType,
+    ReadonlyContentModelBlock,
+    ReadonlyContentModelBlockGroup,
+} from 'roosterjs-content-model-types';
+
+/**
+ * Query content model blocks
+ * @param group The block group to query
+ * @param blockType The type of block to query @default 'Paragraph'
+ * @param filter Optional selector to filter the blocks
+ * @param findFirstOnly True to return the first block only, false to return all blocks
+ */
+export function queryContentModelBlocks<T extends ReadonlyContentModelBlock>(
+    group: ReadonlyContentModelBlockGroup,
+    blockType?: ContentModelBlockType,
```

Comment:
This blockType needs to be coupled with T. Otherwise, you can pass two different types, for example:
```ts
queryContentModelBlock<ContentModelParagraph>(group, 'Table');
```

In theory you should make this fail the build.

Reply:
You can declare like this.
```ts
export function queryContentModelBlocks<T extends ReadonlyContentModelBlock>(
  group: ReadonlyContentModelBlockGroup,
  blockType: T extends ReadonlyContentModelBlockBase<infer U> ? U : never,
 ...) {
   ...
}
```

Reply:
Then blockType and T must match, otherwise build will fail.